glean-core/android/src/main/java/mozilla/telemetry/glean/private/DatetimeMetricType.kt

```diff
@@ -102,8 +102,10 @@ class DatetimeMetricType internal constructor(
                 hour = value.get(Calendar.HOUR_OF_DAY),
                 minute = value.get(Calendar.MINUTE),
                 second = value.get(Calendar.SECOND),
-                nano = value.get(Calendar.MILLISECOND) * 1000000L,
-                offset_seconds = value.get(Calendar.ZONE_OFFSET).div(1000)
+                nano = java.util.concurrent.TimeUnit.MILLISECONDS.toNanos(value.get(Calendar.MILLISECOND).toLong()),
```

Comment:
nit: you could do `import java.util.concurrent.TimeUnit as AndroidTimeUnit` above in the file and then use `AndroidTimeUnit.MILLISECONDS.toNanos` to make this a bit more readable

Reply:
Ha, that was the missing piece.